### PR TITLE
Proper Android 10 support

### PR DESCRIPTION
**Pull request by eliaslecomte**
Starting Android 10, it's no longer possible to connect (as an app) to wifi networks like before.

What can an app still do:

1. Temporary connect to a wifi access point (for example to setup an IoT device)
2. Suggest a network, but the OS can choose what to do with it.

With this PR, i've switched the connection logic to use [WifiUtils](https://github.com/ThanosFisherman/WifiUtils) which just released support for [option 1](https://github.com/ThanosFisherman/WifiUtils/pull/46).

Advantages of using WifiUtils:
* Works well on different android versions
* Only resolves when the app is connected to the network
* Has a good retry mechanism built in
* Well written code

### CODE REVIEW

**Comment by Rapsssito** [comment]
I really like this PR and appreciate your work. However, as discussed in https://github.com/ThanosFisherman/WifiUtils/issues/47, **WifiUtils Android 10 compatibility has not been properly tested**.

@eliaslecomte, have you tested this changes on a real Android device?

File: android/src/main/java/com/reactlibrary/rnwifi/RNWifiModule.java [hunk 1]
```diff
@@ -189,190 +177,41 @@ public void setEnabled(Boolean enabled) {
...
      *
      * @param SSID     name of the network to connect with
      * @param password password of the network to connect with
-     * @param isWep    required for iOS
-     * @param promise
+     * @param isWep    only for iOS
+     * @param promise  to send success/error feedback
      */
     @ReactMethod
     public void connectToProtectedSSID(@NonNull final String SSID, @NonNull final String password, final boolean isWep, final Promise promise) {
```
Comment (Rapsssito) [inline android/src/main/java/com/reactlibrary/rnwifi/RNWifiModule.java]: If this method is now compatible with open WiFi networks, we should document it.

**Comment by eliaslecomte** [comment]
I agree more testing should be done. Specially edge cases. I have tested the happy flow on a Pixel 4.

**Comment by Rapsssito** [comment]
@eliaslecomte, if someone uses this module to connect a network indefinitely, does this PR break that functionality?

**Comment by eliaslecomte** [comment]
> @eliaslecomte, if someone uses this module to connect a network indefinitely, does this PR break that functionality?

I fear for it. I don't see how with the changes for Android 10.
And even when you target Android 9 or previous, the connect to wifi functionality no longer works on Android 10.

**Comment by Rapsssito** [comment]
@eliaslecomte, while this module targets API 28, it will still work on Android 10. I am currently using it on my personal device on Android 10.

The option 2 from your original comment looks more like the scope of this project. The module suggest a network and user decides to connect. This is how iOS handles the connections at the moment.

**Comment by eliaslecomte** [comment]
> I love the simplicity of using this library.
> I know we will lose control a little. (is the trade-off)
> 
> I request add a commit with guidelines of semmantic-release
> `BREAKING CHANGE: [Android] Use com.thanosfisherman.wifiutils to connect to WiFi`

I'll will go through the change requests on thursday.

**Comment by eliaslecomte** [comment]
> @eliaslecomte, while this module targets API 28, it will still work on Android 10. I am currently using it on my personal device on Android 10.
> 
> The option 2 from your original comment looks more like the scope of this project. The module suggest a network and user decides to connect. This is how iOS handles the connections at the moment.

Problem is that even when targetting sdk 28 it won't work on all Android 10 devices. For example it doesn't work on the Samsung s10.

File: android/src/main/java/com/reactlibrary/rnwifi/RNWifiModule.java (hunk 1 above)
Comment (eliaslecomte) [reply android/src/main/java/com/reactlibrary/rnwifi/RNWifiModule.java]: I assume on Android you could always connect to an open network with using an empty string as password. In my opinition we should also have the same connectTo methods on iOS and Android.

**Comment by eliaslecomte** [comment]
I've opened a new PR as I didn't want to rewrite all of the history here (changing commit messages..); https://github.com/JuanSeBestia/react-native-wifi-reborn/pull/46

**Pull request closed without merging**